crates/kitsune_p2p/types/src/config.rs

```diff
@@ -154,10 +176,24 @@ pub mod tuning_params_struct {
         gossip_max_batch_size: u32 = 1_000_000,
 
         /// Should gossip dynamically resize storage arcs?
+        ///
+        /// This is an unstable feature must be enabled with the "unstable-sharding" Cargo feature.
+        /// This setting is not available without the feature.
+        ///
+        /// [Default: true]
+        #[cfg(feature = "unstable-sharding")]
```

Comment:
Tuning params was designed to be tolerant to unknown configs so we could drop them without friction, so this change isn't strictly necessary... But I'm also okay with it because it doesn't add any friction to the end-users or app devs.

Reply:
Yeah, I went back and forth with myself about this one. Compiling it out should affect end users because it'll just be ignored. However, it does mean that we can't reference this in code anywhere without intentionally doing so and dealing with the feature flags. That should make it hard to accidentally write code that adjusts arcs... Or at least that's what I was thinking, only time will test that theory :)